### PR TITLE
Enhance Find Dialog: display extra info in the status bar

### LINKED ISSUE #14307: Find Dialog: display "(Filtered!)" in the status bar
Quite often users are bewildered after executing a search and seeing the result(s).
[A recent example](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/14304).

**Suggestion:**
 Display the range & conditions in the Find Dialog status bar.

***

Notepad++ v8.5.8.

### PR DESCRIPTION
**Pull request by donho**
Fix #14307

### CODE REVIEW

File: PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp [hunk 1]
```diff
@@ -2486,17 +2486,15 @@ bool FindReplaceDlg::processFindNext(const TCHAR *txt2find, const FindOption *op
...
+				wstring reasonMsg;
+				bool isTheMostLaxMode = pOptions->_isWrapAround && !pOptions->_isMatchCase && !pOptions->_isWholeWord;
+				if (!isTheMostLaxMode)
 				{
-					wstring sep = pNativeSpeaker->getLocalizedStrFromID("find-status-cannot-find-separator", L": ");
-					wstring reasonMsg = pNativeSpeaker->getLocalizedStrFromID("find-status-cannot-find-no-wrap", L"turn on \"Wrap around\" maybe?");
-
-					warningMsg += sep;
-					warningMsg += reasonMsg;
+					reasonMsg = pNativeSpeaker->getLocalizedStrFromID("find-status-cannot-find-pebkac-maybe", L"The given occurence cannot be found. You may have forgotten to check \"Wrap around\" (to ON), \"Match case\" (to OFF), or \"Match whole word only\" (to OFF).");
```
Comment (alankilborn) [inline PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:2493]: I like the introduction of `pebkac` into the code! :-)

**Comment by alankilborn** [comment]
I tested this.  

For _Find Next_ and _Find Previous_, it appears to be working well.

It needs to be extended to command button presses that initiate these types of search actions:

* Count
* Replace
* Replace All
* Mark All

The message should appear and should be changed to NOT mention _Wrap around_ for these:

* Find All in Current Document
* Find All in All Opened Documents
* Replace All in All Opened Documents
* Find in Files
* Replace in Files
* Find in Projects
* Replace in Projects

**Comment by donho** [comment]
> The message should appear and should be changed to NOT mention Wrap around for these:

>    Find All in Current Document
    Find All in All Opened Documents
    Replace All in All Opened Documents
    Find in Files
    Replace in Files
    Find in Projects
    Replace in Projects



The initial scope of issue #14307 was just to prevent user from ignoring "Wrap around".
AFAICT, all the PEBCAK concerning this issue is only on "Find Next" (please provide any link if I'm wrong).
Now, I understand you guys' point to keep the consistency of software - I agree to add it for all the commands concerning "Wrap around". But adding all above commands for being aligned the behaviour and for the "constency" is just too much, and not necessary IMO.

**Comment by alankilborn** [comment]
I wonder if it is difficult.  I mean, I'd think most search paths lead to a common point or two (where the error bubble could be put up)...

**Comment by Yaron10** [comment]
@donho,

> Now, I understand you guys' point for keeping the consistency of software

If someone (@alankilborn ? :) ) is willing to work on the other inconsistencies (red/blue color, beep, flash) in a separate PR?

**Comment by alankilborn** [comment]
>If someone (alankilborn ? :) ) is willing to work on the other inconsistencies

I don't know...there has to be more interesting things for me to try my hand at, than that...  :-D

**Comment by Yaron10** [comment]
@alankilborn,

Retracting. :)

**Comment by donho** [comment]
> If someone (@alankilborn ? :) ) is willing to work on the other inconsistencies (red/blue color, beep, flash) in a separate PR?

There's no inconsistency here for me :)

File: PowerEditor/installer/nativeLang/english.xml [hunk 2]
```diff
@@ -1666,6 +1666,7 @@ Find in all files but exclude all folders log or logs recursively:
 			<find-status-replace-not-found value="Replace: no occurrence was found"/>
 			<find-status-replace-readonly value="Replace: Cannot replace text. The current document is read only"/>
 			<find-status-cannot-find value="Find: Can't find the text &quot;$STR_REPLACE$&quot;"/>
+			<find-status-cannot-find-pebkac-maybe value="The given occurence cannot be found. You may have forgotten to check &quot;Wrap around&quot; (to ON), &quot;Match case&quot; (to OFF), or &quot;Match whole word only&quot; (to OFF)."/>
```
Comment (fml2) [inline PowerEditor/installer/nativeLang/english.xml:1669]: "occurence" -- shouldn't it be spelled "occurrence", i.e. 2 x "r" ?

**Pull request closed without merging**